tests/system/src/test/java/test/robot/javafx/scene/RTLTextCharacterIndexTest.java

```diff
@@ -0,0 +1,383 @@
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import test.util.Util;
+
+/*
+ * Test for verifying character index of Text nodes in RTL orientation.
+ *
+ * There are 6 tests in this file.
+ * Here, the scene node orientation is set to RTL for all the tests.
+ * Steps for testTextInfoForRTLEnglishText()
+ * 1. Create a Text node and add it to the scene using a VBox.
+ * 2. Add only english text to the Text node.
+ * 3. Move the cursor from right to left with a random
+ *    decrement value generated in step() method.
+ * 4. Character index should change from highest value to lowest
+ *    as expected.
+ *
+ * Steps for testTextInfoForRTLArabicText()
+ * 1. Create a Text node and add it to the scene using a VBox.
+ * 2. Add only arabic text to the Text node.
+ * 3. Move the cursor from right to left with a random
+ *    decrement value generated in step() method.
+ * 3. Character index should increment as expected since it is RTL text.
+ *
+ * Steps for testTextInfoForRTLEnglishArabicText()
+ * 1. Create a Text node and add it to the scene using a VBox.
+ * 2. Add both english and arabic text to the Text node.
+ * 3. Move the cursor from right to left with a random
+ *    decrement value generated in step() method.
+ * 4. Character index should change in decreasing order for english text
+ *    and in increasing order for arabic text.
+ *
+ * Steps for testTextInfoForMultiLineRTLEnglishText()
+ * 1. Create a Text node and add it to the scene using a VBox.
+ * 2. Add two lines of only english text to the Text node.
+ * 3. Move the cursor from right to left with a random
+ *    decrement value generated in step() method.
+ * 4. Character index should change in decreasing order as expected.
+ *
+ * Steps for testTextInfoForMultiLineRTLEnglishArabicText()
+ * 1. Create a Text node and add it to the scene using a VBox.
+ * 2. Add two lines of both english and arabic text to the Text node.
+ * 3. Move the cursor from right to left with a random
+ *    decrement value generated in step() method.
+ * 4. Character index should change in decreasing order for english text
+ *    and increasing order for arabic text.
+ *
+ * Steps for testTextInfoForMultiLineRTLArabicText()
+ * 1. Create a Text node and add it to the scene using a VBox.
+ * 2. Add two lines of only arabic text to the Text node.
+ * 3. Move the cursor from right to left with a random
+ *    decrement value generated in step() method.
+ * 4. Character index should change in increasing order as expected.
+ */
+
+public class RTLTextCharacterIndexTest {
+    static CountDownLatch startupLatch = new CountDownLatch(1);
+    static Random random;
+    static Robot robot;
+    static Text text;
+    static VBox vBox;
+
+    static volatile Stage stage;
+    static volatile Scene scene;
+
+    static final int WIDTH = 500;
+    static final int HEIGHT = 200;
+
+    final int Y_OFFSET = 30;
+    final int X_LEADING_OFFSET = 10;
```

Comment:
If they're constant, they should also be `static`, otherwise may want to write them with lower case.